src/TourPortal.js

```diff
@@ -290,13 +292,18 @@ class TourPortal extends Component {
   }
 
   maskClickHandler = e => {
-    const { closeWithMask, onRequestClose } = this.props
+    const { closeWithMask, onRequestClose, showCloseButton } = this.props
     if (
       closeWithMask &&
-      !e.target.classList.contains(CN.mask.disableInteraction)
+      !e.target.classList.contains(CN.mask.disableInteraction) ||
+      !showCloseButton
     ) {
       onRequestClose(e)
     }
+    !showCloseButton && !closeWithMask &&
+      console.warn(
+        `You can't use 'closeWithMask' prop when 'showCloseButton' prop is false`
```

Comment:
I'm not sure about this `warn`. 

Could you please expan a little on why? Remember there is also the `esc` keyboard _trigger_ to close the _Tour_

Reply:
As you think, I forgot `esc` keyboard trugger. Thank you.

Reply:
no prob!